// sanity-plugin-content-calendar/build/AgendaEvent.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = AgendaEvent;

var _react = _interopRequireDefault(require("react"));

var _hooks = require("./hooks");

var _preview = _interopRequireDefault(require("part:@sanity/base/preview"));

var _schema = _interopRequireDefault(require("part:@sanity/base/schema"));

var _dateFns = require("date-fns");

var _User = _interopRequireDefault(require("./User"));

var _AgendaEvent = _interopRequireDefault(require("./AgendaEvent.css"));

var _config = require("./config");

var _intent = _interopRequireDefault(require("part:@sanity/components/buttons/intent"));

var _historyIcon = _interopRequireDefault(require("part:@sanity/base/history-icon"));

var _editIcon = _interopRequireDefault(require("part:@sanity/base/edit-icon"));

var _Warning = _interopRequireDefault(require("./Warning"));

var _draftUtils = require("part:@sanity/base/util/draft-utils");

function AgendaEvent(_ref) {
  var event = _ref.event;
  var hasDraft = (0, _hooks.useHasChanges)(event);
  var publishedId = (0, _draftUtils.getPublishedId)(event.doc._id || '');
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].root,
    title: event.title
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].inner
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].event
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].previewContainer
  }, /*#__PURE__*/_react["default"].createElement(_preview["default"], {
    value: event.doc,
    type: _schema["default"].get(event.doc._type),
    layout: "media"
  })), /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].details
  }, /*#__PURE__*/_react["default"].createElement("h3", {
    className: _AgendaEvent["default"].label
  }, "Title"), /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].value
  }, event.title), /*#__PURE__*/_react["default"].createElement("h3", {
    className: _AgendaEvent["default"].label
  }, "Scheduled for"), /*#__PURE__*/_react["default"].createElement("time", {
    className: _AgendaEvent["default"].value
  }, (0, _dateFns.format)(event.start, _config.dateFormat), " \u2022 ", (0, _dateFns.format)(event.start, _config.timeFormat)), event.user && _config.showAuthor && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("h3", {
    className: _AgendaEvent["default"].label
  }, "By"), " ", /*#__PURE__*/_react["default"].createElement(_User["default"], {
    user: event.user
  })))), /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].actionsWrapper
  }, hasDraft && /*#__PURE__*/_react["default"].createElement("div", {
    className: _AgendaEvent["default"].warning
  }, /*#__PURE__*/_react["default"].createElement(_Warning["default"], null)), /*#__PURE__*/_react["default"].createElement(_intent["default"], {
    className: _AgendaEvent["default"].intentButton,
    color: "primary",
    icon: hasDraft ? _historyIcon["default"] : _editIcon["default"],
    intent: "edit",
    params: {
      id: publishedId,
      type: event.doc._type
    }
  }, hasDraft ? 'Review changes' : 'Edit'))));
}
