// sanity-plugin-content-calendar/build/EventDialog.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = EventDialog;

var _react = _interopRequireDefault(require("react"));

var _intent = _interopRequireDefault(require("part:@sanity/components/buttons/intent"));

var _default = _interopRequireDefault(require("part:@sanity/components/buttons/default"));

var _default2 = _interopRequireDefault(require("part:@sanity/components/dialogs/default"));

var _EventDialog = _interopRequireDefault(require("./EventDialog.css"));

var _buttonGrid = _interopRequireDefault(require("part:@sanity/components/buttons/button-grid"));

var _historyIcon = _interopRequireDefault(require("part:@sanity/base/history-icon"));

var _editIcon = _interopRequireDefault(require("part:@sanity/base/edit-icon"));

var _schema = _interopRequireDefault(require("part:@sanity/base/schema"));

var _preview = _interopRequireDefault(require("part:@sanity/base/preview"));

var _hooks = require("./hooks");

var _Warning = _interopRequireDefault(require("./Warning"));

var _dateFns = require("date-fns");

var _User = _interopRequireDefault(require("./User"));

var _config = require("./config");

var _draftUtils = require("part:@sanity/base/util/draft-utils");

function EventDialog(_ref) {
  var event = _ref.event,
      isOpen = _ref.isOpen,
      onClose = _ref.onClose;
  var hasChanges = (0, _hooks.useHasChanges)(event);
  var publishedId = (0, _draftUtils.getPublishedId)(event.doc._id);
  var title = _config.dialogTitle === 'date' ? (0, _dateFns.format)(event.start, _config.dateFormat) : 'Schedule details';
  return /*#__PURE__*/_react["default"].createElement(_default2["default"], {
    isOpen: isOpen,
    title: _config.dialogTitle ? _config.dialogTitle : title,
    onClose: onClose,
    onCloseClick: onClose,
    padding: "none"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].root
  }, hasChanges && /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].warning
  }, /*#__PURE__*/_react["default"].createElement(_Warning["default"], null)), /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].inner
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].previewContainer
  }, /*#__PURE__*/_react["default"].createElement(_preview["default"], {
    value: event.doc,
    type: _schema["default"].get(event.doc._type),
    layout: "media"
  })), /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].details
  }, /*#__PURE__*/_react["default"].createElement("h3", {
    className: _EventDialog["default"].label
  }, "Title"), /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].value
  }, event.title), /*#__PURE__*/_react["default"].createElement("h3", {
    className: _EventDialog["default"].label
  }, "Scheduled for"), /*#__PURE__*/_react["default"].createElement("time", {
    className: _EventDialog["default"].value
  }, (0, _dateFns.format)(event.start, _config.dateFormat), " \u2022 ", (0, _dateFns.format)(event.start, _config.timeFormat)), event.user && _config.showAuthor && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("h3", {
    className: _EventDialog["default"].label
  }, "By"), /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].user
  }, /*#__PURE__*/_react["default"].createElement(_User["default"], {
    user: event.user
  }))))), /*#__PURE__*/_react["default"].createElement("div", {
    className: _EventDialog["default"].actionsWrapper
  }, /*#__PURE__*/_react["default"].createElement(_buttonGrid["default"], {
    align: "end"
  }, /*#__PURE__*/_react["default"].createElement(_intent["default"], {
    color: "primary",
    icon: hasChanges ? _historyIcon["default"] : _editIcon["default"],
    intent: "edit",
    params: {
      id: publishedId,
      type: event.doc._type
    }
  }, hasChanges ? 'Review changes' : 'Edit'), /*#__PURE__*/_react["default"].createElement(_default["default"], {
    kind: "secondary",
    onClick: onClose
  }, "Cancel")))));
}
