// sanity-plugin-content-calendar/build/User.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = User;

var _react = _interopRequireDefault(require("react"));

var _components = require("@sanity/base/components");

var _userColor = require("@sanity/base/user-color");

var _User = _interopRequireDefault(require("./User.css"));

function User(_ref) {
  var user = _ref.user,
      _ref$minimal = _ref.minimal,
      minimal = _ref$minimal === void 0 ? false : _ref$minimal;
  var userColorManager = (0, _userColor.useUserColorManager)();
  var color = userColorManager.get(user.id || null);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: _User["default"].root
  }, minimal ? /*#__PURE__*/_react["default"].createElement("span", null, /*#__PURE__*/_react["default"].createElement(_components.UserAvatar, {
    userId: user.id
  })) : /*#__PURE__*/_react["default"].createElement("div", {
    className: _User["default"].user,
    style: {
      backgroundColor: color.background,
      color: color.text
    }
  }, /*#__PURE__*/_react["default"].createElement("span", null, /*#__PURE__*/_react["default"].createElement(_components.UserAvatar, {
    userId: user.id
  })), /*#__PURE__*/_react["default"].createElement("span", {
    className: _User["default"].displayName
  }, user.displayName)));
}
